tensorflow_datasets/image_classification/visual_domain_decathlon.py

```diff
@@ -107,9 +107,7 @@ def _split_generators(self, dl_manager):
     if self.builder_config.name == 'imagenet12':
       devkit_path, images_archive = dl_manager.download_and_extract([
           _URL_PREFIX_VGG + 'decathlon-1.0-devkit.tar.gz',
-          tfds.download.Resource(
-              url=_URL_PREFIX_IMAGENET + 'decathlon-1.0-data-imagenet.tar',
-              extract_method=tfds.download.ExtractMethod.NO_EXTRACT),
```

Comment:
Removing `NO_EXTRACT` looks like a bug to me. Now `images_archive` will linked to the extracted dir, rather than the archive. I'm quite surprised this is working. Could you investigate why this is not crashing ? Are you sure that you're generating using the modified code ?

Reply:
@Conchylicultor That indeed, looks like a bug. I generated the default dataset during testing using `tfds.load('visual_domain_decathlon')` due to which it executes the else block which puts the path in `image_archive` manually, therefore, preventing the bug from happening.
https://github.com/tensorflow/datasets/blob/c7439f84c2a77891e07a8e154cbde21f6ddcc8e9/tensorflow_datasets/image_classification/visual_domain_decathlon.py#L119
Currently, I do not have the bandwidth required to try downloading the complete `imagenet12` dataset.
However, we could use `tfds.download` instead of `tfds.download_and_extract` in case of `imagenet12` builder config, thus allowing to follow the given program flow

Reply:
I think keeping `NO_EXTRACT` in this case if fine, as this allow to download urls in parallel, while a separate `dl_manager.download` will download sequentially.
Thank you for confirming. I'll only merge the other dataset. So nothing to do from your end.